sql/planbuilder/builder.go

```diff
@@ -1424,26 +1494,23 @@ func (b *PlanBuilder) analyzeOrderBy(fromScope, projScope *scope, order ast.Orde
 					b.handleErr(err)
 				}
 				target := projScope.cols[intIdx-1]
-				var gf *expression.GetField
-				if target.scalar != nil {
-					gf = expression.NewGetFieldWithTable(int(target.id), target.typ, "", target.scalar.String(), target.nullable)
-				} else {
-					gf = expression.NewGetFieldWithTable(int(target.id), target.typ, target.table, target.col, target.nullable)
+				scalar := target.scalar
+				if scalar == nil {
+					scalar = expression.NewGetFieldWithTable(int(target.id), target.typ, target.table, target.col, target.nullable)
 				}
 				outScope.addColumn(scopeColumn{
```

Comment:
Constructor that takes `(target)` then sets scalar / other fields might be useful here

Reply:
added to new PR